src/app/contexts/Migration/Migration.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { ethers, Contract } from "ethers";
+import {
+	ARKMigrationViewStructOutput,
+	Migration,
+	MigrationTransactionStatus,
+} from "@/domains/migration/migration.contracts";
+import { useEnvironmentContext } from "@/app/contexts";
+import { MigrationRepository } from "@/repositories/migration.repository";
+import { useProfileWatcher } from "@/app/hooks/use-profile-watcher";
+
+const CONTRACT_ADDRESS = import.meta.env.VITE_POLYGON_CONTRACT_ADDRESS;
+const POLYGON_RPC_URL = import.meta.env.VITE_POLYGON_RPC_URL;
+
+const contractABI = [
+	{
+		inputs: [
+			{
+				internalType: "bytes32[]",
+				name: "arkTxHashes",
+				type: "bytes32[]",
+			},
+		],
+		name: "getMigrationsByArkTxHash",
+		outputs: [
+			{
+				components: [
+					{
+						internalType: "address",
+						name: "recipient",
+						type: "address",
+					},
+					{
+						internalType: "uint96",
+						name: "amount",
+						type: "uint96",
+					},
+					{
+						internalType: "bytes32",
+						name: "arkTxHash",
+						type: "bytes32",
+					},
+				],
+				internalType: "struct ARKMigrator.ARKMigrationView[]",
+				name: "",
+				type: "tuple[]",
+			},
+		],
+		stateMutability: "view",
+		type: "function",
+	},
+];
+
+// @TODO: make this transactions dynamic, an option is to store/get using the
+// env storage, e.g. env.storage().get("ark-migration-transactions")
+const fakeTransactionsIds = [
+	// Invalid one
+	"0x74a1e612846da4c3da618a77e82e6f8468fd88459662738babce5b6014b023f4",
+	// Valid ones
+	"0x2c00cb87957dd47ee9e6a03210bf2ca46ba7a11156391d3b5960760846de226a",
+	"0x5e4f8256caffe7d7276da0611bcf8a8bff8aea1219853ea07d9575355efff704",
+];
+
+const transactionMapper = (migration: ARKMigrationViewStructOutput): Migration => {
+	const amount = Number.parseFloat(ethers.utils.formatEther(migration.amount.toString()));
+
+	const status =
+		migration.recipient === ethers.constants.AddressZero
+			? MigrationTransactionStatus.Waiting
+			: MigrationTransactionStatus.Confirmed;
+
+	return {
+		// @TODO: get the address and timestamp
+		// Not sure yet how that will work but maybe we can use the profile to get the address
+		// and then the transactions or maybe we can save those values in the env storage
+		// when adding a new migration
+		address: "PeNdInGPeNdInGPeNdInGPeNdInG",
+		amount,
+		id: migration.arkTxHash,
```

Comment:
this should link to the polygon ID instead. Currently the UI links to polygonscan with the ARK transaction ID instead